Makefile

```diff
@@ -15,7 +15,12 @@ PYVER := $(shell $(PYTHON) -c "import sys; print(sys.version_info[0])")
 ROOT_PATH := $(shell pwd)
 
 PYINC := $(shell $(PYCONFIG) --includes)
-PYLIB := $(shell $(PYCONFIG) --ldflags) -L$(shell $(PYCONFIG) --prefix)/lib
+
+PYLIB := $(shell $(PYCONFIG) --ldflags --embed)
+ifneq ($(.SHELLSTATUS), 0)
```

Comment:
Can you use `PYVER` to decide what to do instead of trying, failing retrying?

Reply:
Yes, I've updated it to use PYVER. But, it meant modifying PYVER because we need to test for three situations and is a bit tricksy.  Do say if you have any better ideas.

Reply:
Oh jeez, I thought `PYVER` was `28`, `36` and so on :flushed: But of course make doesn't offer numeric comparisons either. `bc` does though.

This seems to work...

```diff
diff --git a/Makefile b/Makefile
index 517286f..4d4c480 100644
--- a/Makefile
+++ b/Makefile
@@ -10,18 +10,22 @@ PYTHON ?= python
 PYCONFIG ?= python-config
 PY2TO3 ?= 2to3
 
-# PYVER value is 2 or 3
-PYVER := $(shell $(PYTHON) -c "import sys; print(sys.version_info[0])")
+# PYVER is 207, 308 etc.
+PYVER := $(shell $(PYTHON) -c "import sys; print(sys.version_info[0] * 100 + sys.version_info[1])")
 ROOT_PATH := $(shell pwd)
 
 PYINC := $(shell $(PYCONFIG) --includes)
-PYLIB := $(shell $(PYCONFIG) --ldflags) -L$(shell $(PYCONFIG) --prefix)/lib
+ifeq (1,$(shell echo "$(PYVER) >= 308" | bc))
+    PYLIB := $(shell $(PYCONFIG) --ldflags --embed)
+else
+    PYLIB := $(shell $(PYCONFIG) --ldflags) -L$(shell $(PYCONFIG) --prefix)/lib
+endif
 
 BUILD_DIR = build/lib.$(PYVER)
 
 .PHONY: build check py3 clean
 
-ifeq (2,$(PYVER))
+ifeq (1,$(shell echo "$(PYVER) < 300" | bc))
 
 build:
        $(PYTHON) setup.py build --build-lib $(BUILD_DIR)
```

Reply:
Interesting, but thinking from a distribution perspective adding a new dependency is not so great. Even if bc is likely to be installed.

I've had another go at PYVER to create a one-liner that can return 2, 3 or 3embed.